homework/index.js

```diff
@@ -30,14 +32,145 @@
+        const imageUrl = contributors[index].avatar_url;
+        const name = contributors[index].login;
+        const contributionsCount = contributors[index].contributions;
+        const profileURL = contributors[index].html_url;
+        const tr = createAndAppend('div', table, { class: 'clickable row' });
+        tr.onclick = () => openNewTab(profileURL);
+        const td1 = createAndAppend('div', tr, { class: 'col-30' });
+        const imgContainerDiv = createAndAppend('div', td1, {
+          class: 'image-container',
+        });
+        createAndAppend('img', imgContainerDiv, {
+          src: imageUrl,
+          width: 80,
+          height: 80,
+        });
+        createAndAppend('div', tr, { text: name, class: 'col-30 description' });
+        const td3 = createAndAppend('div', tr, { class: 'col-30' });
+        createAndAppend('label', td3, { text: contributionsCount });
+      }
+    });
+  }
+  function bindRepositoryDetails(selectedRepository) {
+    const { description } = selectedRepository.description;
+    const { forks } = selectedRepository.forks;
+    const repositoryName = selectedRepository.name;
+    const updatedOn = new Date(selectedRepository.updated_at).toLocaleString();
+    const repoURL = selectedRepository.html_url;
+
+    const infoContainerDiv = document.getElementById('repo-info-container');
+    const repoInfodiv = createAndAppend('div', infoContainerDiv, {
+      class: 'repo-info',
+    });
+    const div = createAndAppend('div', repoInfodiv);
+    const table = createAndAppend('div', div, { class: 'grid' });
+
+    const tr1 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr1, { text: 'Repository:', class: 'col-50' });
+    const repoNameCol = createAndAppend('div', tr1, { class: 'col-50' });
+    createAndAppend('a', repoNameCol, {
+      href: repoURL,
+      target: '_blank',
+      text: repositoryName,
+    });
+
+    const tr2 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr2, { text: 'Description:', class: 'col-50' });
+    createAndAppend('div', tr2, { text: description, class: 'col-50' });
+
+    const tr3 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr3, { text: 'Forks:', class: 'col-50' });
+    createAndAppend('div', tr3, { text: forks, class: 'col-50' });
+
+    const tr4 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr4, { text: 'Updated:', class: 'col-50' });
+    createAndAppend('div', tr4, { text: updatedOn, class: 'col-50' });
+  }
+
+  function repositoryChanged(event) {
+    const infoElement = document.getElementById('repo-info-container');
+    infoElement.innerHTML = '';
+    const repositoryIndex = parseInt(event.target.selectedOptions[0].value, 10);
+    const selectedRepo = repositoryList[repositoryIndex];
+    bindRepositoryDetails(selectedRepo);
+    bindContributions(selectedRepo.contributors_url);
+  }
```

Comment:
You can pass the selected repo directly as an argument:

```js
  function repositoryChanged(selectedRepo) {
    const infoElement = document.getElementById('repo-info-container');
    infoElement.innerHTML = '';
    // const repositoryIndex = parseInt(event.target.selectedOptions[0].value, 10);
    // const selectedRepo = repositoryList[repositoryIndex];
    bindRepositoryDetails(selectedRepo);
    bindContributions(selectedRepo.contributors_url);
  }
```

See how that works on line 155.

Reply:
done